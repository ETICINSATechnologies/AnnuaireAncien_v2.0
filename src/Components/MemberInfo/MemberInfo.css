.MemberInfo .infoArea {
    display: grid;
    grid-template-rows: 1vw 3.5vw 2.5vw 3.5vw repeat(5, 1.5vw 3.2vw) 0.5vw;
    grid-template-columns: 3fr 1fr 4fr;
    cursor: default;
}

.MemberInfo .infoArea * {
    width: 90%;
    font-size: var(--default-font-size);
    grid-column: 1 / 4;
    vertical-align: middle;
    margin: 0 auto auto auto;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.MemberInfo .infoArea .editPhoto {
    grid-column: 3;
    grid-row: 1;
    height: 90%;
    width: auto;
    padding: 0.7vw;
    margin: auto 0 0 auto;
}

.MemberInfo .infoArea .editPhoto:hover {
    cursor: pointer;
    opacity: 0.8;
}

.MemberInfo .infoArea .memberPhoto {
    grid-row: 1 / 5;
    grid-column: 1;
    width: 90%;
    margin: auto 0 0.6vw auto;
}

.MemberInfo .infoArea .inputLabel {
    font-size: var(--input-label-font-size);
    margin: auto auto 0;
}

.MemberInfo .infoArea input {
    padding: 0.4vw;
    vertical-align: middle;
    margin: 0 auto auto;
}

.MemberInfo .infoArea .modifyButton {
    grid-column: 1 / 3;
}

.MemberInfo .infoArea .boardButton {
    grid-column: 1 / 4;
    border: 0.23vmin solid black;
    border-radius: 1.2vmin;
    width: auto;
    margin: 1vw 2.5vw auto auto;
    padding: 0.2vw 0.8vw;
}

.MemberInfo .infoArea .updateButton {
    border: 0.23vmin solid black;
    border-radius: 1vmin;
    width: 80%;
    margin: auto;
    padding: 0.2vw 0.8vw;
}

.MemberInfo .infoArea input[type='button']:hover {
    cursor: pointer;
    opacity: 0.8;
}

.MemberInfo .infoArea .cancel {
    grid-column: 1 / 3;
    grid-row: auto;
    margin-top: 1vw;
    background-color: #ff3f34;
}

.MemberInfo .infoArea .save {
    grid-column: 3;
    grid-row: auto;
    margin-top: 1vw;
    background: #95c589;
}

.MemberInfo .infoArea .addBoard {
    background: #95c589;
}

.MemberInfo .infoArea .deleteBoard {
    background-color: #ff3f34;
}

.MemberInfo .infoArea .firstName {
    text-align: center;
    font-weight: bolder;
    font-size: var(--name-font-size);
    grid-row: 2;
    grid-column: 2 / 4;
    margin: 1vw auto 0 auto;
}

.MemberInfo .infoArea .lastName {
    text-align: center;
    margin: auto;
    font-weight: bolder;
    font-size: var(--name-font-size);
    grid-row: 3;
    grid-column: 2 / 4;
}

.MemberInfo .infoArea .telephone {
    text-align: center;
    margin: auto;
    font-size: var(--telephone-font-size);
    grid-row: 4;
    grid-column: 2 / 4;
}

.MemberInfo .infoArea .linkImg {
    margin: auto;
    width: 5vmin;
}

.MemberInfo .infoArea .link {
    text-align: center;
    margin: auto 1vmin auto 1vmin;
}

.MemberInfo .infoArea .linkContainer {
    text-align: center;
    margin: auto;
    grid-row : 5;
    grid-column: 2 / 4;
}