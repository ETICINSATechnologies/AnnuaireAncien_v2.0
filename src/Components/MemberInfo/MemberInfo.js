import React, {Component} from 'react';
import './MemberInfo.css';

import defaultMan from '../../Images/default_man.svg'
import defaultWoman from '../../Images/default_woman.svg'
import edit from '../../Images/edit.png'


class MemberInfo extends Component {
    constructor(props) {
        super(props);
        this.propertiesName = {
            firstName: null,
            lastName: null,
            telephone: null,
            email: 'email',
            company: 'travaille chez',
            year: 'dernière année à ETIC',
            position: 'dernier poste à ETIC',
            department: 'département'
        };
        this.state = {
            edition: false,
            isBoard: false,
            properties: null,
        };
    }

    getNewProperty() {
        let newProperties = {};
        Object.keys(this.propertiesName).forEach((property) => {
            if (this.props.info.hasOwnProperty(property)) {
                if (property === 'department')
                    newProperties[property] = this.props.info[property].name;
                else
                    newProperties[property] = this.props.info[property];
            }
        });
        if (this.props.info['positions']) {
            newProperties['position'] = this.props.info['positions'][0].label;
            newProperties['year'] = '2018';
        }

        return newProperties;
    }

    render() {
        let newProperties = this.getNewProperty();
        let propertyDisplay = [];
        Object.keys(newProperties).forEach((property) => {
            if (property === 'firstName' || property === 'lastName' || property === 'telephone') {
                propertyDisplay.push(
                    <div key={property} className={property}>
                        {newProperties[property]}
                    </div>
                )
            }
            else {
                propertyDisplay.push(
                    <div key={`label ${property}`} className={`inputLabel ${property}`}>
                        {this.propertiesName[property]}
                    </div>);
                propertyDisplay.push(
                    <input key={`input ${property}`} type="text" className={property} value={newProperties[property]}
                           disabled={!this.state.edition}/>
                )
            }
        });

        let gender = 'M';
        if (this.props.info['gender'])
            gender = this.props.info['gender'].label;

        return (
            <section className='MemberInfo'>
                <div className='infoTitle'> Info du membre</div>
                {
                    Object.keys(newProperties).length !== 0 ?
                        <div className='infoArea'>
                            <img className='memberPhoto' alt='défaut' title="http://www.onlinewebfonts.com/icon"
                                 src={gender === 'F' ? defaultWoman : defaultMan}/>
                            {propertyDisplay}
                            {
                                this.props.editable ?
                                    <React.Fragment>
                                        <img className='editPhoto' src={edit} alt="modifier"
                                             style={this.state.edition ? {display: 'none'}: {}}
                                             onClick={() => {this.setState({edition: true})}}/>
                                        {
                                            this.state.edition ?
                                                <React.Fragment>
                                                    <input type="button" onClick={() => this.setState({edition: false})}
                                                           value="annuler" className='updateButton cancel'/>
                                                    <input type="button" onClick={() => this.setState({edition: false})}
                                                           value="enregistrer" className='updateButton save'/>
                                                </React.Fragment> :
                                                this.state.isBoard ?
                                                    <input type="button" className='boardButton deleteBoard'
                                                           value="Supprimer du CA"/> :
                                                    <input type="button" className='boardButton addBoard'
                                                           value="Ajouter au CA"/>
                                        }
                                    </React.Fragment> : ''
                            }
                        </div> :
                        ''
                }
            </section>
        );
    }
}

export default MemberInfo;



