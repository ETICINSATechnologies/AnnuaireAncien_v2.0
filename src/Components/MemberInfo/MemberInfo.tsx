import React, {Component} from 'react';

import './MemberInfo.css';
import facebook from '../../Images/facebook.png'
import linkedin from '../../Images/linkedin.png'

import {Member} from "../../Model/Member";


interface MemberInfoProps {
    member?: Member
    image: string
}

const MAIN_PROPERTIES = ["firstName", "lastName", "telephone"];

const LINKS = ["linkedin", "facebook"];

const OTHER_PROPERTIES = {
    email: 'Email',
    latestYear: 'Dernière année à ETIC',
    latestPosition: 'Dernier poste à ETIC',
    birthday: 'Date de naissance',
};

class MemberInfo extends React.Component<MemberInfoProps> {
    render() {
        const linkMaker = (link: string): string => {
            let ret: string = '';
            link ?
                ret = link.indexOf('http://') ? 'http://' + link : link :
                ret = '';

            return ret;
        };

        if (this.props.member) {
            let member = this.props.member.read();

            let mainProperties = MAIN_PROPERTIES.map((property) =>
                <div key={property} className={property} title={member![property]}>{member![property]}</div>
            );
            let linkProperties = LINKS.map((property, index) => {
                    if (member[property])
                        return (
                            <a key={index} className='link' href={member![property]}>
                                <img className='linkImg'
                                     src={property === 'facebook' ? facebook : linkedin} alt={property}/>
                            </a>
                        )
                }
            );

            let otherProperties = Object.keys(OTHER_PROPERTIES).map((property) => {
                if (member[property]) {
                    return (
                        <React.Fragment key={property}>
                            <div className={`inputLabel ${property}`}>{(OTHER_PROPERTIES as any)[property]}</div>
                            <input type="text" className={property} value={member[property]} disabled={true}/>
                        </React.Fragment>
                    )
                }
            });

            return (
                <section className='MemberInfo'>
                    <div className='infoTitle'>Info du membre</div>
                    <div className='infoArea'>
                        <img className='memberPhoto' alt="default" title="http://www.onlinewebfonts.com/icon"
                             src={this.props.image}/>
                        {mainProperties}
                        <div className='linkContainer'>{linkProperties}</div>
                        {otherProperties}
                    </div>
                </section>
            )
        }

        return <section className='MemberInfo'/>
    }
}

export default MemberInfo;



