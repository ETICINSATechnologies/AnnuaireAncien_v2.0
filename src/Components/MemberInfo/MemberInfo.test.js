import React from 'react';
import MemberInfo from './MemberInfo';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


Enzyme.configure({adapter: new Adapter()});

describe( 'MemberInfo component', () => {
    it('renders without crashing', () => {
            const tree = renderer.create(<MemberInfo />).toJSON();
            expect(tree).toMatchSnapshot();
        }
    );

    it('MemberInfo unit test', () => {
        expect(Enzyme.shallow(<MemberInfo/>)
            .find('.MemberInfo')
            .length
        ).toBe(1)
    });

});