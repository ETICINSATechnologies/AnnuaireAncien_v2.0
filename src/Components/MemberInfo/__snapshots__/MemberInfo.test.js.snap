// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberInfo component renders without crashing 1`] = `
<section
  className="MemberInfo"
>
  <div
    className="infoTitle"
  >
     Info du membre
  </div>
  
</section>
`;
